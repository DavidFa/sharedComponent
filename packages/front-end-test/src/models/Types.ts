export interface PostType {
    id: number;
    title: string;
    body: string
}

export enum PostStatus {
    list,
    edit,
    add
}