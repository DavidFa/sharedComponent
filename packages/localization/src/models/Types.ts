export type InputFormModel = {
    name: InputFormProps;
    description: InputFormProps;
}


export type InputFormProps = {
    label: string;
    placeholder: string;
    length: number;
}

export type InputFormWithLanguageProps = {
    language: string;
    placeholder: string;
    length: number;
}

export interface LocalizationState {
    language: string;
    languages: string[];
}

export enum LocalizationActionType {
    updateLanguage,
    populateLanguages
}

export type LocalizationAction = {
    type: LocalizationActionType;
    payload: {
        language?: string;
        languages?: string[];
    }
}