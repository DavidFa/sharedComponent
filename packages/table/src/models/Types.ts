export type DataGridColumns = {
    field: string;
    headerName: string;
    width: number;
    align?: string;
    sortable?: boolean;
}

export type DataGridRows = {
    [key: string]: string | number
}