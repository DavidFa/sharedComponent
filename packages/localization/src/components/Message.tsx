import React from "react";
import styled from "styled-components";
import { useAppSelector } from "../hooks/hooks";
import { Status } from "../models/Types";


const Wrapper = styled.div`
text-align:center;
color: ${props => props.color}
`;


const Message: React.FC = () => {
    const message = useAppSelector(state => state.message);
    let color = message.status === Status.fail ? 'red' : message.status === Status.success ? "green" : "";

    return <Wrapper color={color}>{message.message}</Wrapper>;
}

export default Message;