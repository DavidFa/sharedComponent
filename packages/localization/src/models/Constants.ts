export const FIELD_NAME: string = "Name";
export const FIELD_DESC: string = "Description";