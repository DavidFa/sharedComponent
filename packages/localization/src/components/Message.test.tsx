import React from "react";
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Message from "./Message";
import { Status } from "../models/Types";

const mockStore = configureStore([]);

describe('React-Redux Message Component', () => {
    let store;
    let component: renderer.ReactTestRenderer;

    beforeEach(() => {
        store = mockStore({
            message: {
                loading: false,
                status: Status.init,
                message: "",
            }
        });

        store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <Message />
            </Provider>
        );
    });

    test('should render with given init status state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})


describe('React-Redux Message Component', () => {
    let store;
    let component: renderer.ReactTestRenderer;

    beforeEach(() => {
        store = mockStore({
            message: {
                loading: false,
                status: Status.success,
                message: "success",
            }
        });

        store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <Message />
            </Provider>
        );
    });

    test('should render with given success status state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})

describe('React-Redux Message Component', () => {
    let store;
    let component: renderer.ReactTestRenderer;

    beforeEach(() => {
        store = mockStore({
            message: {
                loading: false,
                status: Status.fail,
                message: "success",
            }
        });

        store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <Message />
            </Provider>
        );
    });

    test('should render with given fail status state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });
})

