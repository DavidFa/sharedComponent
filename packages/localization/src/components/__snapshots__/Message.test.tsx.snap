// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React-Redux Message Component should render with given fail status state from Redux store 1`] = `
<div
  className="sc-bdfBQB bUlqGe"
  color="red"
>
  success
</div>
`;

exports[`React-Redux Message Component should render with given init status state from Redux store 1`] = `
<div
  className="sc-bdfBQB WFNB"
  color=""
>
  
</div>
`;

exports[`React-Redux Message Component should render with given success status state from Redux store 1`] = `
<div
  className="sc-bdfBQB cKljzO"
  color="green"
>
  success
</div>
`;
